sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -945,13 +976,16 @@ pi_result _pi_queue::executeCommandList(ze_command_list_handle_t ZeCommandList,
     this->ZeOpenCommandListFence = nullptr;
     this->ZeOpenCommandListSize = 0;
   }
-
-  bool UseCopyEngine = getZeCommandListIsCopyList(ZeCommandList);
+  int Index = -1;
+  bool UseCopyEngine = getZeCommandListIsCopyList(ZeCommandList, Index);
   if (UseCopyEngine)
     zePrint("Command list to be executed on copy engine\n");
+  // If available, get the copy command queue assosciated with
+  // ZeCommandList
+  auto ZeCopyCommandQueue =
+      (Index == -1) ? nullptr : ZeCopyCommandQueues[Index];
```

Comment:
Can't you replace getZeCommandListIsCopyList with getCopyQueueIndex and remove the earlier?

Reply:
Done. Good optimization.